spec.emu

```diff
@@ -102,56 +162,60 @@ contributors: Nicolò Ribaudo
+        <emu-alg>
+          1. Return *false*.
+        </emu-alg>
+      </emu-clause>
 
-        <emu-clause id="sec-module-namespace-exotic-objects-get-p-receiver-ReadyForSyncExecution" type="abstract operation">
-          <h1>
-            <ins>
-              ReadyForSyncExecution (
-                _module_: a Cyclic Module Record,
-                optional _seen_: a List of Module Records,
-              ): a Boolean
-            </ins>
-          </h1>
-          <dl class="header"></dl>
-          <emu-alg>
-            1. If _seen_ is not provided, let _seen_ be a new empty List.
-            1. If _seen_ contains _module_, return *true*.
-            1. Append _module_ to _seen_.
-            1. If _module_.[[Status]] is ~evaluated~, return *true*.
-            1. If _module_.[[Status]] is ~evaluating~ or ~evaluating-async~, return *false*.
-            1. Assert: _module_.[[Status]] is ~linked~.
-            1. If _module_.[[HasTLA]] is *true*, return *false*.
-            1. For each ModuleRequest Record _required_ of _module_.[[RequestedModules]], do
-              1. Let _requiredModule_ be GetImportedModule(_module_, _required_.[[Specifier]]).
-              1. If ReadyForSyncExecution(_requiredModule_, _seen_) is *false*, then
-                1. Return *false*.
-            1. Return *true*.
-          </emu-alg>
-        </emu-clause>
+      <emu-clause id="sec-module-namespace-exotic-objects-delete-p" type="internal method">
+        <h1>
+          [[Delete]] (
+            _P_: a property key,
+          ): a normal completion containing a Boolean
+        </h1>
+        <dl class="header">
+          <dt>for</dt>
+          <dd>a module namespace exotic object _O_</dd>
+        </dl>
+        <emu-alg>
+          1. If <del>_P_ is a Symbol</del><ins>IsSymbolLikeNamespaceKey(_P_, _O_)</ins>, return ! OrdinaryDelete(_O_, _P_).
+          1. Let _exports_ be <del>_O_.[[Exports]]</del><ins>? GetModuleExportsList(_O_)</ins>.
+          1. If _exports_ contains _P_, return *false*.
+          1. Return *true*.
+        </emu-alg>
       </emu-clause>
 
-      <emu-clause id="sec-modulenamespacecreate" type="abstract operation" number="10">
+      <emu-clause id="sec-module-namespace-exotic-objects-ownpropertykeys" type="internal method">
+        <h1>[[OwnPropertyKeys]] ( ): a normal completion containing a List of property keys</h1>
+        <dl class="header">
+          <dt>for</dt>
+          <dd>a module namespace exotic object _O_</dd>
+        </dl>
+        <emu-alg>
+          1. Let _exports_ be <del>_O_.[[Exports]]</del><ins>? GetModuleExportsList(_O_)</ins>.
+          1. <ins>If _O_.[[Deferred]] is *true*, and _exports_ contains *"then"*, then</ins>
+            1. <ins>Set _exports_ to a copy of _exports_.</ins>
+            1. <ins>Remove *"then"* from _exports_.</ins>
+          1. Let _symbolKeys_ be OrdinaryOwnPropertyKeys(_O_).
+          1. Return the list-concatenation of _exports_ and _symbolKeys_.
```

Comment:
_symbolKeys_ should be renamed to something else since it may contain a "then" property